src/key_gen/mod.rs

```diff
@@ -415,26 +439,50 @@ impl KeyGen {
                 updated = false;
             }
         }
-        msgs
+
+        let mut unhandables = Vec::new();
+        for msg in cache_messages {
+            let sender = if let Some(name) = self.node_id_from_index(msg.creator()) {
+                name
+            } else {
+                warn!(
+                    "cannot get name of index {:?} among {:?}",
+                    msg.creator(),
+                    self.names
+                );
+                continue;
+            };
+            unhandables.push((sender, msg));
+        }
+        (msgs, unhandables)
     }
 
     fn process_message<R: RngCore>(
         &mut self,
         rng: &mut R,
         msg: Message,
     ) -> Result<Vec<Message>, Error> {
-        debug!(
-            "{:?} with phase {:?} handle DKG message {:?}",
-            self, self.phase, msg
+        trace!(
+            "{:?} with phase {:?} handle DKG message {:?}-{:?}",
+            self,
+            self.phase,
+            msg.id(),
+            msg
         );
-        match msg {
+        match msg.clone() {
             Message::Initialization {
                 key_gen_id,
                 m,
                 n,
                 member_list,
-            } => self.handle_initialization(rng, m, n, key_gen_id, member_list),
-            Message::Proposal { key_gen_id, part } => self.handle_proposal(key_gen_id, part),
+            } => {
+                let _ = self.caching_messages.insert(msg.id(), msg);
```

Comment:
Is this not us caching an incoming message here? 

I thought `caching_messages` was to be used for messages _we_ created?

Reply:
We cache messages that we created and messages that come from other participants as well. Both types flow through this function.